connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Herder.java

```diff
@@ -178,7 +178,17 @@
      * Validate the provided connector config values against the configuration definition.
      * @param connectorConfig the provided connector config values
      */
-    ConfigInfos validateConnectorConfig(Map<String, String> connectorConfig);
+    default ConfigInfos validateConnectorConfig(Map<String, String> connectorConfig) {
+        return validateConnectorConfig(connectorConfig, true);
+    }
+
+    /**
+     * Validate the provided connector config values against the configuration definition.
+     * @param connectorConfig the provided connector config values
+     * @param doLog set it to true to log all connectorConfig with INFO level. false to log nothing.
+     *              Noted: there are many endpoints requiring this method but not all configs are worth being logged.
+     */
+    ConfigInfos validateConnectorConfig(Map<String, String> connectorConfig, boolean doLog);
```

Comment:
Actually, it's the new method that needs a `default` implementation if we choose to add it to this interface.

Reply:
Thanks for suggestions! 

Just curious, is Herder.java a public interface?

Reply:
I trace the history of Herder.java. It seems to me Herder.java is not public interface so it should be fine to add new methods directly. For example, the [commit](https://github.com/apache/kafka/commit/7746301c2643e0037bff3bcb84c41d512937e4c9) added three new methods without default implementation.

Am I missing something here?

Reply:
First, to your first question, no this interface is not public api, but as a general principle we are trying to maintain compatibility regardless, wherever it makes sense. Here, keeping both methods makes the most sense. 

Now, w/r/t the commit that you mention, indeed there's a difference. 
That commit was the implementation of KIP-558. And that's how we document changes to public interfaces, by writing and discussing publicly KIPs. You may read more here: https://cwiki.apache.org/confluence/display/KAFKA/Kafka+Improvement+Proposals

In this PR a KIP would be an overkill, and given that this is not public interface, we can easily maintain compatibility the way I described above. The other difference is that here a default implementation makes sense, while in KIP-558 it didn't make much sense. 

Hope this helps.

Reply:
@kkonstantine thanks for the great explanation!!! I have addressed your comments in the latest commit. Please take a look. thanks.